tests/expected/function-contract/history/noclone_fail.expected

```diff
@@ -0,0 +1,4 @@
+error[E0382]: borrow of moved value: `ptr_renamed`
```

Comment:
Humm... maybe name this something like `kani_old_<var_name>`, so it's a bit more intuitive to the user what's going on.

Another possibility is to create a dummy function:
```rust
fn remember<T: Copy>(val: T) -> T {
  val
}
```

and change the generated code to be:
```rust
let ptr_renamed = kani::internal::remember(ptr);
```

This way, the error will be that `ptr` does not implement Copy trait.

Reply:
The naming is actually because of the renaming of variables within the ensures clause
For this particular example,
```rust
#[kani::ensures(|result| old(ptr).0 + 1 == ptr.0)]
```
becomes
```rust
let ptr_renamed = kani::internal::untracked_deref(&ptr);
let result_kani_internal = function_call(ptr);
kani::assert((|result| old(ptr_renamed).0 + 1 == ptr_renamed.0)(result_kani_internal))
std::mem::forget(ptr_renamed)
```
So the "_renamed" comes from the renaming the ensures statement which is something tangential to this PR, but a plausible solution is provided in #3234. 

And then what this PR is implementing is lifting "old" up to the top
```rust
let ptr_renamed = kani::internal::untracked_deref(&ptr);
let remember_kani_internal_0 = ptr_renamed;
let result_kani_internal = function_call(ptr);
kani::assert((|result| remember_kani_internal_0.0 + 1 == ptr_renamed.0)(result_kani_internal))
std::mem::forget(ptr_renamed)
```
This specific error statement comes from the fact that ownership of `ptr_renamed` gets given to the `remember_kani_internal`, but then we try to access it again via the assert statement which causes the borrow of moved value error at the assert.

This error could be solved by making yet another copy of the `let ptr_for_remember = kani::internal::untracked_deref(&ptr);` which essentially functions as contraction within rust affine's logic. This would allow us to remember non-copyable/non-cloneable data, because the move into the `remember_kani_internal_0` wouldn't clash with borrows within the ensures statement later.

Reply:
decided to do a shuffling around of code so that it uses the name `ptr` which is tracked by rust and gives a nicer error message